### ISSUE TITLE
 Windows Phone : Access is Denied when call method "subscribe" from socket

### ISSUE THREAD

**Issue by charpin** [Atmosphere/atmosphere-javascript#184]
Hello,

I'm currently developing a mobile application and a web application. Both applications are designed to communicate with each other through a chat discussion. As a client who wants to talk to his advisor. 

I'm developing my mobile application on Windows Phone (WP 8.1) with Cordova tool (version 4.0.0) and the Ionic Framework (Version 1.1.0). I use the Visual Studio IDE 2015 RC Community.

I use the 2.2.7 version of Atmosphere-JS.

The problem I have on Windows Phone does not occur on Android or iOS.

The first time I created a socket connection, worked well. As soon as I leave the chat room, it is actually destroy the socket and I create a new socket connection and have the following exception :

Error: Access is denied.
   at connectors.windowref (ms-appx://fr.caindosuez.moninfomandats/www/bower_components/atmosphere/atmosphere.js:708:25)
   at _local (ms-appx://fr.caindosuez.moninfomandats/www/bower_components/atmosphere/atmosphere.js:803:17)
   at _execute (ms-appx://fr.caindosuez.moninfomandats/www/bower_components/atmosphere/atmosphere.js:611:21)
   at execute (ms-appx://fr.caindosuez.moninfomandats/www/bower_components/atmosphere/atmosphere.js:2907:17)
   at atmosphere.subscribe (ms-appx://fr.caindosuez.moninfomandats/www/bower_components/atmosphere/atmosphere.js:2969:9)
   at Anonymous function (ms-appx://fr.caindosuez.moninfomandats/www/app/contact/chat/chatRequest.ctrl.js:54:10)

This following line in atmosphere.js when exception occur :

```
     windowref: function () {
            var win = window.open("", name.replace(/\W/g, ""));
```

This my code from the angular controller to chat request :

```
 function ChatRequest(...) {
   //this controller allow to send and receive chat notification from web application
   var transport = 'long-polling';
   var socket = atmosphere;
   var subSocket;    
   var idChatRoom;

  socket.unsubscribe(); //when page reload, destroy socket connection

  var request = {
    url: Config.serverAddress + '/chat/all', //chat notification
    contentType : "application/json",
    trackMessageLength : true,
    shared : true,
    transport : transport ,
    fallbackTransport: 'long-polling'
  };

  $interval(function(){
    if(subSocket == null)
        subSocket = socket.subscribe(request); //exception occur at this line
  }.bind(this), 2000);  

  request.onOpen = function(response) {
    connected = true;
    if(window.console) {
        console.log("User is connected to the chat room")
    }
 };

 request.onTransportFailure = function(errorMsg, request) {;
    if(window.console) {
        console.log("Error : " + errorMsg)
    } 
  };

  request.onMessage = function (response) {
    var message = response.responseBody;
    try {
        var json = atmosphere.util.parseJSON(message);
    } catch (e) {
        return;
    }
    var typeEvent = json.typeEvent;
    if("newChat" == typeEvent) {
        var object = json.object;
        var message = object.message;
        var idChatRoomSend = object.idChatRoom;

        if("accepted" == message && idChatRoomSend == idChatRoom) {
            if(window.console) {
                console.log("Advisor has accepted the request chat from client");
            }
            accepted = true;
           //destroy socket and subsocket because chatSession will create two new socket and subsocket
            subSocket.disconnect();
            socket.unsubscribe();
            $state.transitionTo("modules.chatSession", {id_chat_room : idChatRoom},{reload:true});
            }
        }
    };

  request.onClose = function(response) {
    connected = false;
    if(window.console) {
        console.log("Close chat room : " + idChatRoom);
    }
  };
```

} 

This controller is the equivalent of my second controller which manage the communication betwen two users (chatSession controller). When one of this user left the chat room, in chatSession controller, I just disconnect the subsocket :

```
            subSocket.disconnect();
            $state.transitionTo("modules.chatRequest", '',{reload:true});
```

When "transitionTo" is called, I reload the chatRequest controller to create a new socket connection. But at this moment, I have the exception "Access is denied". 

In short, this a diagram of chat room is working :
1. User from mobile app -> request for tchat
2. Reception of tchat request by web application -> advisor accept the tchat request 
3. Message send thought the socket "chat/all" to indicate that the tchat request is accepted 
4. Reception of message indicate the tchat request is accepted by mobile application
5. Transition to ChatSession controller to create a new socket for the specific tchat (identified by number)

Do you have any idea how I can solve this problem ?

If you need more details to understand my problem, don't hesitate to answer ;).
